src/e2e/puppeteer/__tests__/toggleSidebar.ts

```diff
@@ -0,0 +1,31 @@
+import click from '../helpers/click'
+import getSelection from '../helpers/getSelection'
+import paste from '../helpers/paste'
+import waitForEditable from '../helpers/waitForEditable'
+import { page } from '../setup'
+
+vi.setConfig({ testTimeout: 20000, hookTimeout: 20000 })
+
+describe('all platforms', () => {
+  it('on toggleSidebar, the caret offset should be preserved once it closes.', async () => {
+    const cursor = 'Hello'
+    const importText = `
+    - ${cursor}`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable('Hello')
+    await click(editableNodeHandle, { offset: 1 })
+
+    await click('[aria-label=menu]')
+    await page.waitForSelector('[aria-label=sidebar] .fade-enter-done')
```

Comment:
`.fade-enter-done` is an animation class that we probably don't want to couple our test to. Can you say more about this?

Relatedly, tests must be only select based on `aria-label` or `data-testid`. This allows us to separate behavior from styles.